File: src/main/java/at/hannibal2/skyhanni/utils/jsonobjects/TrophyFishJson.java
package at.hannibal2.skyhanni.utils.jsonobjects;

import at.hannibal2.skyhanni.features.fishing.trophy.TrophyRarity;
import com.google.gson.annotations.Expose;

import java.util.Map;

public class TrophyFishJson {
    @Expose
    public Map<String, TrophyFishInfo> trophy_fish;

    public static class TrophyFishInfo {
        @Expose
        public String displayName;

        @Expose
        public String description;

        @Expose
        public Integer rate;

        @Expose
        public Map<TrophyRarity, Integer> fillet;
    }
}

File: src/main/java/at/hannibal2/skyhanni/data/jsonobjects/repo/TrophyFishJson.kt
package at.hannibal2.skyhanni.data.jsonobjects.repo

import at.hannibal2.skyhanni.features.fishing.trophy.TrophyRarity
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

data class TrophyFishJson(
    @Expose @SerializedName("trophy_fish") val trophyFish: Map<String, TrophyFishInfo>,
)

data class TrophyFishInfo(
    @Expose val displayName: String,
    @Expose val description: String,
    @Expose val rate: Int?,
    @Expose val fillet: Map<TrophyRarity, Int>,
)
